core/src/main/scala/io/github/gaelrenoux/tranzactio/DatabaseOps.scala

```diff
@@ -58,6 +70,18 @@ trait DatabaseOps[Connection, R0] {
   )(implicit errorStrategies: ErrorStrategiesRef = ErrorStrategies.Parent, trace: Trace): ZIO[R with R0, E, A] =
     transactionOrDie[R, E, A](zio, commitOnFailure)
 
+  final def transactionOrDieStream[R, E, A](
+      stream: => ZStream[Connection with R, E, A],
+      commitOnFailure: => Boolean = false
+  )(implicit errorStrategies: ErrorStrategiesRef = ErrorStrategies.Parent, trace: Trace): ZStream[R with R0, E, A] =
+    transactionStream[R, E, A](stream, commitOnFailure)
+      .mapErrorCause { cause =>
+          cause.flatMap {
+            case Left(dbError) => Cause.die(dbError, cause.trace)
+            case Right(error) => Cause.fail(error, cause.trace)
+          }
+      }
```

Comment:
At first I wondered why you just didn't use `flatMapError` here as on the other methods, then found out it doesn't exist on `ZStream`. However, I can change the existing error-handling on the other methods to match yours (I don't like having two slightly-different error-handling mechanisms, and I feel that yours is better).

I'm not sure what replacing the `flatMapError` may the `mapErrorCause` on other methods would change, any idea?

Reply:
I think the flatMapError is a wider operation you could execute some ZIO for error handling, the mapErrorCause is narrower its only can be used to map the Cause  into an other, so it could not execute ZIO. So I think replacing it would not change anything, because you are only using it to map the Cause into another